manifests/repo.pp

```diff
@@ -45,7 +47,9 @@
   # $baseurl is used in template and takes a form such as
   # http://yum.domain.tld/customrepo/5/8/dev/x86_64
   if $baseurl == 'UNSET' {
-    if $username != undef and $password != undef {
+    if $mirrorlist {
```

Comment:
use the following as it is more specific and easier to read.

`if $mirrorlist != undef`